tests/unit/parsers/test.manifestjson.js

```diff
@@ -3601,4 +3601,125 @@ describe('ManifestJSONParser', () => {
       );
     });
   });
+
+  describe('install_origins', () => {
+    function testInstallOrigins(origins, expectValid) {
+      const addonLinter = new Linter({ _: ['bar'] });
+      const json = validManifestJSON({
+        browser_specific_settings: {
+          gecko: {
+            id: 'foo@bar',
+            install_origins: origins,
+          },
+        },
+      });
+      const manifestJSONParser = new ManifestJSONParser(
+        json,
+        addonLinter.collector
+      );
+      expect(manifestJSONParser.isValid).toEqual(expectValid);
+      return addonLinter.collector;
+    }
+
+    it('allows install origins with valid origin', () => {
+      testInstallOrigins(['https://example.com'], true);
+    });
+
+    it('allows install origins with subdomain origin', () => {
+      testInstallOrigins(['https://foo.example.com'], true);
+    });
+
+    it('allows install origins with punnycode origin', () => {
+      testInstallOrigins(['https://xn--fo-9ja.com'], true);
+    });
+
+    it('allows install origins with IDN origin', () => {
+      testInstallOrigins(['https://foo.bar.栃木.jp'], true);
```

Comment:
Side note: it would be nice to somehow make it easier for us to keep at least some of the test cases that we will have to run on both Firefox and the addons-linter sides in sync between them.

Sharing the entire test case is not really doable, but may be worth moving the list of test case into arrays (in both the Firefox tests and these tests) and just make sure the list is as in sync as it can be (especially in terms of cases being tested).

Reply:
Yeah. Kinda similar to your `it.each` suggestion below, the issue with having arrays containing a bunch of invalid/valid values is that when it's invalid, the reason can be different, and we want to check it... Not sure if it's worth doing several arrays of invalid values, one for each kind of error...

Reply:
Not sure exactly if that's what you're looking for but we can totally do:

```
it.each([
  { origin: 'xyz', expectedError: /some error/ },
  { origin: 'qbc', expectedError: /some other error/ },
])() => {
});
```

Reply:
Sure, I can do that. Note that if we want to share the test values then Firefox should also have the expected error code in its tests.